src/device-config.ts

```diff
@@ -0,0 +1,512 @@
+			varType: 'int',
+			defaultValue: '',
+		},
+		deltaRetryCount: {
+			envVarName: 'SUPERVISOR_DELTA_RETRY_COUNT',
+			varType: 'int',
+			defaultValue: '30',
+		},
+		deltaRetryInterval: {
+			envVarName: 'SUPERVISOR_DELTA_RETRY_INTERVAL',
+			varType: 'int',
+			defaultValue: '10000',
+		},
+		deltaVersion: {
+			envVarName: 'SUPERVISOR_DELTA_VERSION',
+			varType: 'int',
+			defaultValue: '2',
+		},
+		lockOverride: {
+			envVarName: 'SUPERVISOR_OVERRIDE_LOCK',
+			varType: 'bool',
+			defaultValue: 'false',
+		},
+		persistentLogging: {
+			envVarName: 'SUPERVISOR_PERSISTENT_LOGGING',
+			varType: 'bool',
+			defaultValue: 'false',
+			rebootRequired: true,
+		},
+	};
+
+	static validKeys = ['SUPERVISOR_VPN_CONTROL', 'OVERRRIDE_LOCK'].concat(
+		_.map(DeviceConfig.configKeys, 'envVarName'),
+	);
+
+	public constructor({ db, config, logger }: DeviceConfigConstructOpts) {
+		this.db = db;
+		this.config = config;
+		this.logger = logger;
+
+		this.actionExecutors = {
+			changeConfig: async step => {
+				try {
+					if (step.humanReadableTarget) {
+						this.logger.logConfigChange(step.humanReadableTarget);
+					}
+					if (!_.isObject(step.target)) {
+						throw new Error('Non-dictionary value passed to changeConfig');
+					}
+					await this.config.set(step.target as Dictionary<string>);
+					if (step.humanReadableTarget) {
+						this.logger.logConfigChange(step.humanReadableTarget, {
+							success: true,
+						});
+					}
+					if (step.rebootRequired) {
+						this.rebootRequired = true;
+					}
+				} catch (err) {
+					if (step.humanReadableTarget) {
+						this.logger.logConfigChange(step.humanReadableTarget, {
+							err,
+						});
+					}
+					throw err;
+				}
+			},
+			setVPNEnabled: async (step, opts = {}) => {
+				const { initial = false } = opts;
+				if (!_.isString(step.target)) {
+					throw new Error('Non-string value passed to setVPNEnabled');
+				}
+				const logValue = { SUPERVISOR_VPN_CONTROL: step.target };
+				if (!initial) {
+					this.logger.logConfigChange(logValue);
+				}
+				try {
+					await this.setVPNEnabled(step.target);
+					if (!initial) {
+						this.logger.logConfigChange(logValue, { success: true });
+					}
+				} catch (err) {
+					this.logger.logConfigChange(logValue, { err });
+					throw err;
+				}
+			},
+			setBootConfig: async step => {
+				const configBackend = await this.getConfigBackend();
+				if (!_.isObject(step.target)) {
+					throw new Error(
+						'Non-dictionary passed to DeviceConfig.setBootConfig',
+					);
+				}
+				await this.setBootConfig(configBackend, step.target as Dictionary<
+					string
+				>);
+			},
+		};
+	}
+
+	private async getConfigBackend() {
+		if (this.configBackend != null) {
+			return this.configBackend;
+		}
+		const dt = await this.config.get('deviceType');
+		if (!_.isString(dt)) {
+			throw new Error('Could not detect device type');
+		}
+
+		return (this.configBackend = configUtils.getConfigBackend(dt) || null);
```

Comment:
Could you separate this to a line for the assignment and a line for the return please